README.md

```diff
@@ -157,24 +157,25 @@ The signature is as follows:
 
 ```typescript
 type EqualityComparator = (a: any, b: any, meta?: any) => boolean;
+type ExtendedEqualityComparator = (a: any, b: any, indexOrKey?: any, parentA?: any, parentB?: any, meta?: any) => boolean;
```

Comment:
Rather than have two types of equality comparators, I believe a consistent one should be used. It would involve a small change internally, but it makes the custom comparator use-case more flexible.

Reply:
If you mean a single type for every comparator, I think it would be confusing. The comparator returned by `createCustomEqual` would have indexOrKey, parentA, parentB, which are not supposed to be user provided. The example in the docs would become :
```ts
const isDeepEqualOrFooMatchesMeta = createCustomEqual(
  (deepEqual) => (objectA, objectB, indexOrKey, parentA, parentB, meta) =>
    objectA.foo === meta ||
    objectB.foo === meta ||
    deepEqual(objectA, objectB, indexOrKey, parentA, parentB, meta)
);
isDeepEqualOrFooMatchesMeta({ foo: 'bar' }, { foo: 'baz' },  undefined, undefined, undefined, 'bar')
```

Another way we could do it is to have all internal comparators a single type and only a different type for the "external" comparator (the return of `createComparator`) ? It would allow us to simplify the `EqualityComparatorCreator` type and thus the docs :
```ts
export type EqualityComparatorCreator = (fn: EqualityComparator) => EqualityComparator;
```
But it would mean having the user type `indexOrKey`, `parentA`, `parentB` in the `deepEqual` call if he uses meta, when it's not necessary (since those properties are only for the custom comparator and not used in `comparator`) :
```ts
const isDeepEqualOrFooMatchesMeta = createCustomEqual(
  (deepEqual) => (objectA, objectB, indexOrKey, parentA, parentB, meta) =>
    deepEqual(objectA, objectB, indexOrKey, parentA, parentB, meta), // <-- Here
);
```
Plus, I think it will be confusing that the `deepEqual` passed as an argument doesn't take the same arguments as `isDeepEqualOrFooMatchesMeta`.
And it would mean still having two types : one for the methods exposed by the package and one for the internal logic.

I'm not sure about this... I think the current way it's done is the less confusing one. 
Maybe I didn't understand what you had in mind.

Reply:
Ah yes, as you lay out the usage what you describe makes sense, in terms of the two separate contracts. Based on this, my only recommendation then would be to rename the "extended" comparator to instead be the "internal" comparator, to better explain the different uses.

Reply:
I agree, it' difficult to find a good name for it...
"Extended" only conveys that there is something more to it than the other one, but not what it's used for.
"Internal" is better since it conveys where it's used. Even though the user interact with, it's as a returned function. The user provides the comparator that will be used internally. I think it could be an ok name.
```ts
type EqualityComparatorCreator = (EqualityComparator) => InternalEqualityComparator;
```
I think that having `IsEqual` and `EqualityComparator` will be confusing. It is the same thing expressed differently. Without context, choosing between the two for typing `deepEqual` would be impossible. Plus, `isEqual` is the name of many other libraries export (like lodash).
I think "internal" is the one that provides the most information, even though it's not perfect.